Cython/Compiler/Nodes.py

```diff
@@ -7757,6 +7773,10 @@ def analyse_declarations(self, env):
             self.else_clause.analyse_declarations(env)
 
     def analyse_expressions(self, env):
+        dummy = object()
+        in_try_block = getattr(env, "in_try_block", dummy)
+        env.in_try_block = True
```

Comment:
I'm well aware that this isn't the right way to do it.

This feels like it should be done in a transform somewhere but it isn't obvious which one is right. I'm just putting doing this to test the idea.